agent_pool.go

```diff
@@ -214,6 +217,28 @@ func (s *agentPools) Delete(ctx context.Context, agentPoolID string) error {
 	return s.client.do(ctx, req, nil)
 }
 
+// List IPs by agent pool ID.
+func (s *agentPools) ListAgentPoolIPs(ctx context.Context, agentPoolID string) (*AgentPoolList, error) {
+	if !validStringID(&agentPoolID) {
+		return nil, ErrInvalidAgentPoolID
+	}
+
+	u := fmt.Sprintf("agent-pools/%s/agents", url.QueryEscape(agentPoolID))
+	req, err := s.client.newRequest("GET", u, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	poolList := &AgentPoolList{}
```

Comment:
The IP address attribute mapping should be added to the `AgentPool` type in `AgentPoolList` since the intent is to retrieve the IPs. Good progress 🔥